cmd/snap/cmd_run.go

```diff
@@ -116,8 +116,19 @@ func getSnapInfo(snapName string, revision snap.Revision) (*snap.Info, error) {
 // returns the environment that is important for the later stages of execution
 // (like SNAP_REVISION that snap-exec requires to work)
 func snapExecEnv(info *snap.Info) []string {
+	home := os.Getenv("HOME")
+	// HOME is not set for systemd services, so pull it out of passwd
+	if home == "" {
+		user, err := user.Current()
+		if err == nil {
+			home = user.HomeDir
+		}
```

Comment:
If there **is** an error, should we bail here? Otherwise it defines `$SNAP_USER_DATA=snap/foo/bar`, which snap-confine won't accept anyway. It'd be nice to fail with a nicer error message than snap-confine gives you.